### PR TITLE
[Azure] Switch to dockerhub

### PR DESCRIPTION
**Pull request by stanislavlevin**
`registry.fedoraproject.org/f32/fedora-toolbox` image is used to build packages on Azure Pipelines.
registry.fedoraproject.org experiences an availability problem and makes unstable FreeIPA CI.

Fedora also distributes its official images on https://hub.docker.com/_/fedora.
`fedora:32` is already used by FreeIPA CI to build the image for tests.

### CODE REVIEW

File: ipatests/azure/Dockerfiles/Dockerfile.build.fedora [hunk 1]
```diff
@@ -4,10 +4,6 @@ ENV container=docker LANG=en_US.utf8 LANGUAGE=en_US.utf8 LC_ALL=en_US.utf8
 
 ADD dist /root
 RUN echo 'deltarpm = false' >> /etc/dnf/dnf.conf \
-    && dnf install -y dnf-plugins-core \
-    && dnf config-manager --set-enabled updates-testing \
-    && dnf config-manager --set-enabled updates-testing-modular \
-    && dnf config-manager --set-disabled fedora-cisco-openh264 \
```
Comment (fcami) [inline ipatests/azure/Dockerfiles/Dockerfile.build.fedora:10]: Why is this change necessary? Is this because the images differ?

Comment (abbra) [reply ipatests/azure/Dockerfiles/Dockerfile.build.fedora:10]: this seems to be unnecessary with the fedora:32 image anymore.

**Comment by abbra** [comment]
Azure CI pipeline: https://dev.azure.com/freeipa/freeipa/_build/results?buildId=5674&view=results
It moved past build and lint stages into running IPA tests so we can consider this change successful.

**Comment by stanislavlevin** [comment]
Please, wait a bit, the keyrings are not skipped. Let me check why.

File: ipatests/azure/Dockerfiles/Dockerfile.build.fedora (hunk 1 above)
Comment (stanislavlevin) [reply ipatests/azure/Dockerfiles/Dockerfile.build.fedora:10]: That was added by me in 26f96595b010396f055c7d11a645be57fc637863 :
> Azure: Make dnf repos consistent
    Build container(image registry.fedoraproject.org/f32/fedora-toolbox)
    has two more dnf repos enabled compared to Tests container(image
    fedora:32). This results in the packages built within the Build
    container can have dependencies which are unresolvable(missing)
    within Tests container.

With this change `build` stage and `tests` one will use the same parent image.

**Comment by stanislavlevin** [comment]
heh, container within VM is not detected for some reason:
```
[root@97684d414c4b /]# /usr/bin/systemd-detect-virt --container
none
[root@97684d414c4b /]# echo $?
1
```

**Comment by stanislavlevin** [comment]
Ah, https://github.com/systemd/systemd/blob/master/src/basic/virt.c#L522
```
        /* Fallback for cases where PID 1 was not systemd (for example, cases where init=/bin/sh is used. */
        r = getenv_for_pid(1, "container", &m);
        if (r > 0) {
                e = m;
                goto translate_name;
        }
```

File: ipatests/azure/Dockerfiles/Dockerfile.build.fedora (hunk 1 above)
Comment (fcami) [reply ipatests/azure/Dockerfiles/Dockerfile.build.fedora:10]: ack for this, thanks for the explanation.

**Comment by abbra** [comment]
master:

* 2b85bfb03052b56591b06abe62c71e36170dc9a0 Azure: Switch to dockerhub provider
* e5c09675f3c7c2ecb105df4c4ecf846a2d3ffdc4 ipatests: Skip keyring tests on containerized platforms

**Comment by stanislavlevin** [comment]
So, to let `systemd-detect-virt` to work correctly one should run this util as a privileged user ( to be able to read
`/proc/1/environ`), otherwise read `/run/systemd/container` \[0\], which is populated by `systemd` on startup.
But Azure run tasks within containers as an unprivileged user and the `build` image, in turn, comes without pre-installed `systemd` package. Thus, to let `systemd` to be the PID 1 we have to re-build container image, which takes additional time if made during CI or pre-made which is additional maintenance duty.

Thus, the best option is to leave as-is :-)

\[0\]: https://www.freedesktop.org/wiki/Software/systemd/ContainerInterface/
